File: pinyougou-user-interface/src/main/java/com/pinyougou/user/UserService.java
package com.pinyougou.user;

import com.pinyougou.pojo.TbUser;

public interface UserService {
    //获取短信验证码
    void getValidCode(String phone);

    //比较用户输入的验证码和生成的验证码是否一致
    boolean verify(String phone,String validCode);

    //用户注册
    void addUser(TbUser user);
}

File: pinyougou-user-web/src/main/webapp/js/service/userService.js
app.service("userService",function ($http) {
    //通过手机号请求后台发送验证码
    this.getValidCode=(phone)=>{
        alert("调用service层生成验证码");
        return $http.get("./user/getValidCode.do?phone="+phone);
    }
    //到后台进行注册
    this.reg=(entity,validCode)=>{
        return $http.post("./user/reg.do?validCode="+validCode,entity);
    }
})